### PR TITLE
Add support for in-memory file system

### PR DESCRIPTION
**Pull request by tomaszalesak**

### CODE REVIEW

File: Lib/Translation/TranslationDb.cs [hunk 1]
```diff
@@ -209,28 +210,70 @@ public void SaveLangDbs(string dir, bool justUsed)
...
-            jw.WriteEndArray();
-        }
+                    jw.WriteEndArray();
+                }
 
-        jw.WriteEndArray();
-        jw.Flush();
+                jw.WriteEndArray();
+                jw.Flush();
+                _fsAbstraction.WriteAllUtf8(PathUtils.Join(dir, "locations.json"), Encoding.UTF8.GetString(stream.ToArray()));
```
Comment (Bobris) [inline Lib/Translation/TranslationDb.cs:265]: It is in UTF-8 already, no need to double decode it and encoded.
Also it is ok to use this "MemoryStream" implementation for all fsAbstraction implementation => not duplicated code is better eventhough it could be teoreticaly less optimal.

File: Shared/DiskCache/FakeFsAbstraction.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using Shared.Utils;
+using Shared.Watcher;
+
+namespace Shared.DiskCache;
+
+public class FakeFsAbstraction : IFsAbstraction, IDirectoryWatcher
```
Comment (Bobris) [inline Shared/DiskCache/FakeFsAbstraction.cs:10]: Rename it to InMemoryFs

**Review by Bobris — CHANGES_REQUESTED** [review]
Generally I would not have anything against merging "Shared" back to "Lib".

File: Lib/Translation/TranslationDb.cs (hunk 1 above)
Comment (tomaszalesak) [reply Lib/Translation/TranslationDb.cs]: Fixed. In InMemoryFs I chose to save files as object type. It is converted to string/byte[] as soon as it is read.

File: Lib/Utils/CommandLineParser/Parser/CommandLineCommand.cs [hunk 3]
```diff
@@ -106,7 +106,7 @@ string[] ParseOneArgument(string[] args, List<CommandLineArgument> arguments)
     {
         var firstArg = args[0].Trim();
         var argument = arguments.FirstOrDefault(a =>
-            a.Words?.Contains(firstArg) ?? !firstArg.StartsWith('-'));
+            a.Words?.Contains(firstArg) ?? !firstArg.StartsWith("-"));
```
Comment (Bobris) [inline Lib/Utils/CommandLineParser/Parser/CommandLineCommand.cs:109]: Why this change from char to string?

File: Lib/Utils/CommandLineParser/Parser/CommandLineParser.cs [hunk 4]
```diff
@@ -24,7 +24,7 @@ public static CommandLineCommand Parse(string[] args, List<CommandLineCommand> c
 
         // first argument
         string firstArg = args?.Length > 0 ? args[0].Trim() : null;
-        if (firstArg != null && !firstArg.StartsWith('-'))
+        if (firstArg != null && !firstArg.StartsWith("-"))
```
Comment (Bobris) [inline Lib/Utils/CommandLineParser/Parser/CommandLineParser.cs:27]: Why change it?

**Review by Bobris — APPROVED** [review]

File: Lib/Utils/CommandLineParser/Parser/CommandLineParser.cs (hunk 4 above)
Comment (tomaszalesak) [reply Lib/Utils/CommandLineParser/Parser/CommandLineParser.cs]: Fixed.

File: Lib/Utils/CommandLineParser/Parser/CommandLineCommand.cs (hunk 3 above)
Comment (tomaszalesak) [reply Lib/Utils/CommandLineParser/Parser/CommandLineCommand.cs]: Probably IDE's reformat. Fixed.
